### DISCUSSION TITLE
Get the Settings of the PartDefinition (Textbox)

### DISCUSSION THREAD

**Discussion by vitalybrandes** [OrchardCMS/OrchardCore#10719] [Q&A] [answered]
I have textbox defined as list with predefined values
How can i access this values from code to build dropdown list in headless UI

**Comment by Skrypt** [comment]
Use IContentDefinitionManager service.

**Reply by vitalybrandes** [reply]
This is what i found, question what next ? :-)
```
var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(nameof(Features));
var contentPart = contentTypeDefinition.Parts.FirstOrDefault(c => c.PartDefinition.Name == nameof(Features));
var Part = contentPart.PartDefinition.Fields.FirstOrDefault(x => x.Name == "Name");
```
@Skrypt , what i do next ?

**Reply by Skrypt** [reply]
```C#
   var partFieldDefinition = _contentDefinitionManager.GetPartDefinition(Part)?.Fields.FirstOrDefault(f => f.Name == field);
   var fieldSettings = partFieldDefinition?.GetSettings<TextFieldPredefinedListEditorSettings>();
```

It is maybe not exactly these vars but you get the idea.

**Comment by vitalybrandes** [comment, marked as answer]
> ```cs
>    var partFieldDefinition = _contentDefinitionManager.GetPartDefinition(Part)?.Fields.FirstOrDefault(f => f.Name == field);
>    var fieldSettings = partFieldDefinition?.GetSettings<TextFieldPredefinedListEditorSettings>();
> ```
> 
> It is maybe not exactly these vars but you get the idea.

Thank you!!!
